uefi/src/proto/shell/mod.rs

```diff
@@ -2,12 +2,141 @@
 
 //! EFI Shell Protocol v2.2
 
-use crate::proto::unsafe_protocol;
+#![cfg(feature = "alloc")]
+
+use alloc::vec::Vec;
+use uefi_macros::unsafe_protocol;
+use uefi_raw::Status;
+
+use core::ptr;
 
 pub use uefi_raw::protocol::shell::ShellProtocol;
 
+use crate::{CStr16, Char16};
+
 /// Shell Protocol
 #[derive(Debug)]
 #[repr(transparent)]
 #[unsafe_protocol(uefi_raw::protocol::shell::ShellProtocol::GUID)]
 pub struct Shell(uefi_raw::protocol::shell::ShellProtocol);
+
+impl Shell {
+    /// Gets the environment variable or list of environment variables
+    ///
+    /// # Arguments
+    ///
+    /// * `name` - The environment variable name of which to retrieve the
+    ///   value
+    ///   If None, will return all defined shell environment
+    ///   variables
+    ///
+    /// # Returns
+    ///
+    /// * `Some(Vec<env_value>)` - Value of the environment variable
+    /// * `Some(Vec<env_names>)` - Vector of environment variable names
+    /// * `None` - Environment variable doesn't exist
+    #[must_use]
+    pub fn get_env<'a>(&'a self, name: Option<&CStr16>) -> Option<Vec<&'a CStr16>> {
```

Comment:
Please don't export this unintuitive/unrusty UEFI API in this high-level wrapper. Instead, please aim to stay close to `std`. This means:
- https://doc.rust-lang.org/proc_macro/tracked_env/fn.var.html (with `&CStr16)
- https://doc.rust-lang.org/std/env/fn.vars.html )

Reply:
Hi Philipp, thanks again for the resources. However, it is not clear to me what you are asking. Are you asking for ``get_env()`` to have a similar return type to the ``var()`` and ``vars()`` functions? Or are you asking for ``get_env()`` to be split up into two separate functions- one dedicated to returning the value of an environment variable and the other dedicated to returning the current environment variables?

Reply:
Sorry for the confusion!

> Or are you asking for get_env() to be split up into two separate functions- one dedicated to returning the value of an environment variable and the other dedicated to returning the current environment variables?

This.

Reply:
No worries! I can see how having separate dedicated functions is more intuitive. In the spirit of mirroring ``std``, I have updated the protocol with these two new functions using the same naming convention as ``var`` and ``vars`` where:
``get_env()`` - retrieves the value of an environment variable
``get_envs()`` - retrieves the names of all environment variables